operators/src/processing/temporal_raster_aggregation.rs

```diff
@@ -0,0 +1,1074 @@
+                        query,
+                        self.tiling_specification,
+                        ctx,
+                        spec,
+                    );
+
+                    Ok(s.boxed())
+                } else {
+                    let spec = TemporalRasterAggregationSubQuery {
+                        fold_fn: fold_future::<P, MinAccFunction>,
+                        no_data_value: self.no_data_value,
+                        initial_value: P::max_value(),
+                        step: self.window,
+                    };
+
+                    let s = RasterOverlapAdapter::<'a, P, _, _>::new(
+                        &self.source,
+                        query,
+                        self.tiling_specification,
+                        ctx,
+                        spec,
+                    );
+
+                    Ok(s.boxed())
+                }
+            }
+            AggregationType::Max => {
+                if self.ignore_no_data {
+                    let spec = TemporalRasterAggregationSubQuery {
+                        fold_fn: no_data_ignoring_fold_future::<P, MaxIgnoreNoDataAccFunction>,
+                        no_data_value: self.no_data_value,
+                        initial_value: P::min_value(),
+                        step: self.window,
+                    };
+
+                    let s = RasterOverlapAdapter::<'a, P, _, _>::new(
+                        &self.source,
+                        query,
+                        self.tiling_specification,
+                        ctx,
+                        spec,
+                    );
+
+                    Ok(s.boxed())
+                } else {
+                    let spec = TemporalRasterAggregationSubQuery {
+                        fold_fn: fold_future::<P, MaxAccFunction>,
+                        no_data_value: self.no_data_value,
+                        initial_value: P::min_value(),
+                        step: self.window,
+                    };
+
+                    let s = RasterOverlapAdapter::<'a, P, _, _>::new(
+                        &self.source,
+                        query,
+                        self.tiling_specification,
+                        ctx,
+                        spec,
+                    );
+
+                    Ok(s.boxed())
+                }
+            }
+            AggregationType::Mean => Err(error::Error::NotYetImplemented), // TODO: implement
+        }
+    }
+}
+
+pub trait AccFunction {
+    /// produce new accumulator value from current state and new value
+    fn acc<T: Pixel>(no_data: Option<T>, acc: T, value: T) -> T;
+}
+pub trait NoDataIgnoringAccFunction {
+    /// produce new accumulator value from current state and new value, ignoring no data values
+    fn acc_ignore_no_data<T: Pixel>(no_data: Option<T>, acc: T, value: T) -> T;
+}
+
+struct MinAccFunction {}
+
+impl AccFunction for MinAccFunction {
+    fn acc<T: Pixel>(no_data: Option<T>, acc: T, value: T) -> T {
```

Comment:
Shouldn't be the no_data values rather in the struct than in every call?

Reply:
it is a static method so we can't access data in the struct